addons/point_of_sale/models/pos_config.py

```diff
@@ -463,10 +463,30 @@ def _force_http(self):
             return True
         return False
 
-    def _get_pos_base_url(self):
-        return '/pos/web' if self._force_http() else '/pos/ui'
-
     # Methods to open the POS
+    def _action_to_open_ui(self):
+        if not self.current_session_id:
+            self.env['pos.session'].create({'user_id': self.env.uid, 'config_id': self.id})
+        path = '/pos/web' if self._force_http() else '/pos/ui'
+        return {
+            'type': 'ir.actions.act_url',
+            'url': path + '?config_id=%d' % self.id,
+            'target': 'self',
+        }
+
+    def _check_before_creating_new_session(self):
+        self._check_pricelists()
+        self._check_company_journal()
+        self._check_company_invoice_journal()
+        self._check_company_payment()
+        self._check_currencies()
+        self._check_profit_loss_cash_journal()
+        self._check_payment_method_ids()
+
+    def _check_before_opening_ui(self):
```

Comment:
why creating this funuction? and not just call _validate_fields?

Reply:
probably a leftover of my previous implementation